src/application.vala

```diff
@@ -232,24 +224,109 @@ public class Application : Gtk.Application {
         }
     }
 
-    private void show_qrcode() {
+    private void create_cert() {
+        GnuTLS.X509.PrivateKey key = GnuTLS.X509.PrivateKey.create();
+        key.generate(GnuTLS.PKAlgorithm.RSA, 2048);
+        //GnuTLS.X509.Certificate certi = GnuTLS.X509.Certificate.create();
+    }
+
+    private string? call_sytem_function (string[] args) {
         try {
-            string read;
-            FileUtils.get_contents ("server.pem", out read);
+            string[] spawn_args = args;
+            string[] spawn_env = Environ.get ();
+            string ls_stdout;
+            string ls_stderr;
+            int ls_status;
+
+            Process.spawn_sync (".",
+                                spawn_args,
+                                spawn_env,
+                                SpawnFlags.SEARCH_PATH,
+                                null,
+                                out ls_stdout,
+                                out ls_stderr,
+                                out ls_status);
+            /*
+            stdout.printf ("stdout:\n");
+            stdout.puts (ls_stdout);
+            stdout.printf ("stderr:\n");
+            stdout.puts (ls_stderr);
+            stdout.printf ("status: %d\n", ls_status);*/ 
+            return ls_stdout;
+
+        } catch (SpawnError e) {
+            stdout.printf ("Error: %s\n", e.message);
+            return null;
+        }
+    }
+
+    private void create_cert_bash() {
+        var path_to_conf = GLib.Environment.get_user_config_dir() + "/nuntius/";
+        var file = File.new_for_path (path_to_conf + "nuntius.pem");
+        if (file.query_exists ()) {
+            string[] spawn_args = {"./createcert.sh"};
+            call_sytem_function(spawn_args);
+            cert_created = true;
+        }
+        try {
+            cert = new TlsCertificate.from_files (path_to_conf + "nuntius.pem", path_to_conf + "nuntius.key");
         } catch (Error e) {
-            warning("error reading file");
+            warning("Failed to load certificate: %s", e.message);
         }
     }
 
+    private string get_fingerprint() {
+        string path_to_pem = GLib.Environment.get_user_config_dir() + "/nuntius/nuntius.pem";
+        string[] spawn_args = {"openssl", "x509", "-in", path_to_pem, "-fingerprint", "-sha1"};
+        string result = call_sytem_function(spawn_args); 
+        string[] splitted = result.split("=");
+        splitted = splitted[1].split("\n");
+        string fingerprint = splitted[0];
+        return fingerprint;
+    }
+
+    private string get_hostname() {
+        string[] spawn_args = {"hostname"};
+        string hostname = call_sytem_function(spawn_args); 
+        return hostname;
+    }
+
+    private void show_qrcode() {
+        var dialog = new Gtk.Dialog.with_buttons ("Certificate fingerprint",
```

Comment:
Use proper capitalization: "Certificate Fingerprint". Beside the string should be marked for translation.
The code is also not aligned here